### PR TITLE
db maintenance auto

### PR DESCRIPTION
**Pull request by bloodearnest**
- refactor: add a start_thread method in service.py
- chore: update dev tooling due to broken pip/pip-tools
- chore: add fixture for mocking subrocess.run
- feat: support detecting and activating maintenance mode
- fix: communicate current mode in job-server requests

### CODE REVIEW

File: jobrunner/sync.py [hunk 1]
```diff
@@ -70,7 +70,10 @@ def api_request(method, path, *args, **kwargs):
     url = "{}/{}/".format(config.JOB_SERVER_ENDPOINT.rstrip("/"), path.strip("/"))
     # We could do this just once on import, but it makes changing the config in
     # tests more fiddly
```
Comment (evansd) [inline jobrunner/sync.py:72]: micronit: this comment is now false as we now rely on setting the headers per-request
```suggestion
```

**Review by evansd — APPROVED** [review]
Looks great. Just one question about what we call the config flag.

File: jobrunner/config.py [hunk 2]
```diff
@@ -184,6 +184,14 @@ def parse_job_resource_weights(config_file):
     STATS_DATABASE_FILE = Path(STATS_DATABASE_FILE)
 
 STATS_POLL_INTERVAL = float(os.environ.get("STATS_POLL_INTERVAL", "10"))
+MAINTENANCE_POLL_INTERVAL = float(
+    os.environ.get("MAINTENANCE_POLL_INTERVAL", "300")
+)  # 5 min
+MAINTENANCE_MODE = os.environ.get("MAINTENANCE_MODE", "").lower() in (
```
[the remark this answers is not in the document]
Comment (bloodearnest) [reply jobrunner/config.py:190]: `ENABLE_MAINTENANCE_MODE_THREAD`?

[the remark this answers is not in the document]
Comment (evansd) [reply jobrunner/config.py:190]: Yep, perfect. Happy with anything that distinguishes it from maintenance mode itself.

**Pull request merged**